api/desecapi/models.py

```diff
@@ -195,209 +168,66 @@ def pdns_id(self):
-                # If there are associated RRs, let's save the RRset. This does
-                # not save the RRs yet.
-                rrsets_with_new_rrs.append(rrset)
-                rrset.save()
-
-            # In either case, send a request to pdns so that we can take
-            # advantage of pdns' type validation check (even if no RRs given).
-            rrsets_for_pdns[rrset] = rrsets[rrset]
-
-        # Repeat lock to make sure new RRsets are also locked
-        rrset_qs.filter(q_meaty).select_for_update()
-
-        # Delete empty RRsets
-        qs_empty.delete()
-
-        # Update contents of modified RRsets
-        RR.objects.filter(rrset__in=rrsets_with_new_rrs).delete()
-        RR.objects.bulk_create([rr
-                                for (rrset, rrs) in rrsets_for_pdns.items()
-                                if rrs and rrset in rrsets_with_new_rrs
-                                for rr in rrs])
-
-        # Update published timestamp on domain
-        self.published = timezone.now()
-        self.save()
-
-        # Send RRsets to pdns
-        if rrsets_for_pdns:
-            pdns.set_rrsets(self, rrsets_for_pdns)
-
-        # Return RRsets
-        return list(rrsets_to_return.values())
-
-    @transaction.atomic
-    def delete(self, *args, **kwargs):
-        # Delete delegation if domain is under our registry
-        subname, parent_domain = self.name.split('.', 1)
-        if parent_domain in settings.LOCAL_PUBLIC_SUFFIXES:
-            try:
-                parent = Domain.objects.get(name=parent_domain)
-            except Domain.DoesNotExist:
-                pass
-            else:
-                rrsets = parent.rrset_set.filter(subname=subname,
-                                                 type__in=['NS', 'DS']).all()
-                parent.write_rrsets({rrset: [] for rrset in rrsets})
-
-        # Delete domain
-        super().delete(*args, **kwargs)
-        pdns.delete_zone(self)
-
     @transaction.atomic
     def save(self, *args, **kwargs):
-        new = self.pk is None
         self.clean()
-        self.clean_fields()
+        self.clean_fields()  # TODO this validates the name (?). Does it more? Needed? Remove?
         super().save(*args, **kwargs)
 
-        if new and not self.owner.locked:
-            self.create_on_pdns()
+    def delegate(self):  # TODO change name to something more precise
+        subname = self.name.split('.')[0]
+        parent = Domain.objects.get(name=self.parent_domain_name)
+        keys = self.keys
+        if not keys:
+            raise APIException('Cannot delegate zone %s to %s, as it currently has no keys.' %
+                               (self.name, self.parent_domain_name))
+        RRset.objects.create(
+            subname=subname, type='NS',
+            ttl=3600, contents=settings.DEFAULT_NS,
+            domain=parent,
+        )  # TODO different NS records?
+        RRset.objects.create(
+            subname=subname, type='DS', ttl=60,
```

Comment:
`ttl=300` will go a bit easier on our servers

Reply:
This should be in sync with whatever pdns does

Reply:
What do you mean? pdns does not have a default ttl for DS records, and does what we tell it to do.